### PR TITLE
update the upgrade method of kubelet / kubectl of worker nodes

### LINKED ISSUE #14870: Inadequate kubeadm-upgrade-1-14.md (apt-mark) and Overview of kubeadm
**This is a Bug Report**

**Problem:**
There is a deficiency in the operation of the Ubuntu tab.
https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14/#Upgrade-kubelet-and-kubectl

```bash
$ apt-get install -y kubelet=1.14.3-00 kubectl=1.14.3-00
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following held packages will be changed:
  kubectl kubelet
The following packages will be upgraded:
  kubectl kubelet
2 upgraded, 0 newly installed, 0 to remove and 19 not upgraded.
E: Held packages were changed and -y was used without --allow-change-held-packages.
```

and there seems to be no link on how to upgrade from 1.13 to 1.14.
https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-upgrade/#kubeadm-upgrade-guidance

**Proposed Solution:**

I think that it is good to write the processing of apt-mark like this.
https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14/#upgrade-kubeadm

```bash
$ apt-mark unhold kubelet kubectl && \
>    apt-get update && apt-get install -y kubelet=1.14.3-00 kubectl=1.14.3-00 && \
>    apt-mark hold kubelet kubectl
Canceled hold on kubelet.
Canceled hold on kubectl.
Hit:1 https://download.docker.com/linux/ubuntu xenial InRelease
Hit:2 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu xenial InRelease
Hit:4 https://apt.kubernetes.io kubernetes-xenial InRelease
Hit:5 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:6 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages will be upgraded:
  kubectl kubelet
2 upgraded, 0 newly installed, 0 to remove and 19 not upgraded.
Need to get 30.4 MB of archives.
After this operation, 18.7 MB of additional disk space will be used.
Get:1 https://apt.kubernetes.io kubernetes-xenial/main amd64 kubectl amd64 1.14.3-00 [8,809 kB]
Get:2 https://apt.kubernetes.io kubernetes-xenial/main amd64 kubelet amd64 1.14.3-00 [21.6 MB]
Fetched 30.4 MB in 26s (1,166 kB/s)
(Reading database ... 54633 files and directories currently installed.)
Preparing to unpack .../kubectl_1.14.3-00_amd64.deb ...
Unpacking kubectl (1.14.3-00) over (1.13.7-00) ...
Preparing to unpack .../kubelet_1.14.3-00_amd64.deb ...
Unpacking kubelet (1.14.3-00) over (1.13.7-00) ...
Setting up kubectl (1.14.3-00) ...
Setting up kubelet (1.14.3-00) ...
kubelet set on hold.
kubectl set on hold.
```

```diff
diff --git a/content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md b/content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md
index 6df1d9d0a..014ac8638 100644
--- a/content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md
+++ b/content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md
@@ -316,8 +316,9 @@ without compromising the minimum required capacity for running your workloads.
     {{< tabs name="k8s_kubelet_and_kubectl" >}}
     {{% tab name="Ubuntu, Debian or HypriotOS" %}}
     # replace x in 1.14.x-00 with the latest patch version
-    apt-get update
-    apt-get install -y kubelet=1.14.x-00 kubectl=1.14.x-00
+    apt-mark unhold kubelet kubectl && \
+    apt-get update && apt-get install -y kubelet=1.14.x-00 kubectl=1.14.x-00 && \
+    apt-mark hold kubelet kubectl
     {{% /tab %}}
     {{% tab name="CentOS, RHEL or Fedora" %}}
     # replace x in 1.14.x-0 with the latest patch version
```

Add a link with how to upgrade it from 1.13 to 1.14.

**Page to Update:**
https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14/#upgrade-kubelet-and-kubectl
https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-upgrade/#kubeadm-upgrade-guidance

**Kubernetes Version:**
v1.14.3

### PR DESCRIPTION
**Pull request by 8398a7**
Fixes #14870

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md [hunk 1]
```diff
@@ -316,8 +316,9 @@ without compromising the minimum required capacity for running your workloads.
     {{< tabs name="k8s_kubelet_and_kubectl" >}}
     {{% tab name="Ubuntu, Debian or HypriotOS" %}}
     # replace x in 1.14.x-00 with the latest patch version
-    apt-get update
-    apt-get install -y kubelet=1.14.x-00 kubectl=1.14.x-00
+    apt-mark unhold kubelet kubectl && \
+    apt-get update && apt-get install -y kubelet=1.14.x-00 kubectl=1.14.x-00 && \
+    apt-mark hold kubelet kubectl
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md]: would you care updating the debian/ubuntu tabs on older upgrade docs too?

Comment (8398a7) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md]: I thought that I would fix it together if there was an error.
However, there was no mistake in the old document itself, but should I format the old document according to the present form?

**Comment by 8398a7** [comment]
Look at the [deployed page](https://deploy-preview-14871--kubernetes-io-master-staging.netlify.com/docs/reference/setup-tools/kubeadm/kubeadm-upgrade/#kubeadm-upgrade-guidance)
I'm worried that the guidance does not contain 1.13 to 1.14 method.
Should I fix this as well?
Or is it better to have another PR?

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md (hunk 1 above)
Comment (neolit123) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-14.md]: ok, no need to change the old docs in that case.

**Comment by neolit123** [comment]
> Should I fix this as well?

same PR is ok, in a separate commit.

**Comment by 8398a7** [comment]
> same PR is ok, in a separate commit.

ok, thanks.

**Comment by 8398a7** [comment]
I added the method of upgrade from 1.13 to 1.14 to the content of the #14870 .

**Pull request merged**